File: GTTS/INL.VettingService.Models/VettingBatch_Item.cs
using System;
using System.Collections.Generic;

namespace INL.VettingService.Models
{
    public class VettingBatch_Item : IVettingBatch_Item
    {
        public int Ordinal { get; set; }
        public long VettingBatchID { get; set; }
        public string VettingBatchName { get; set; }
        public int VettingBatchNumber { get; set; }
        public int VettingBatchOrdinal { get; set; }
        public long? TrainingEventID { get; set; }
        public int? CountryID { get; set; }
        public string CountryName { get; set; }
        public int VettingBatchTypeID { get; set; }
        public string VettingBatchType { get; set; }
        public int? AssignedToAppUserID { get; set; }
        public string AssignedToAppUserFirstName { get; set; }
        public string AssignedToAppUserLastName { get; set; }
        public int VettingBatchStatusID { get; set; }
        public string VettingBatchStatus { get; set; }
        public bool IsCorrectionRequired { get; set; }
        public bool? CourtesyVettingOverrideFlag { get; set; }
        public string CourtesyVettingOverrideReason { get; set; }
        public string GTTSTrackingNumber { get; set; }
        public string LeahyTrackingNumber { get; set; }
        public string INKTrackingNumber { get; set; }
        public DateTime? DateVettingResultsNeededBy { get; set; }
        public DateTime? DateSubmitted { get; set; }
        public DateTime? DateAccepted { get; set; }
        public DateTime? DateSentToCourtesy { get; set; }
        public DateTime? DateCourtesyCompleted { get; set; }
        public DateTime? DateSentToLeahy { get; set; }
        public DateTime? DateLeahyResultsReceived { get; set; }
        public DateTime? DateVettingResultsNotified { get; set; }
        public DateTime? EventStartDate { get; set; }
        public DateTime? DateLeahyFileGenerated { get; set; }
        public int VettingFundingSourceID { get; set; }
        public string VettingFundingSource { get; set; }
        public int AuthorizingLawID { get; set; }
        public string AuthorizingLaw { get; set; }
        public string VettingBatchNotes { get; set; }
        public int ModifiedByAppUserID { get; set; }
        public DateTime ModifiedDate { get; set; }
        public string TrainingEventName { get; set; }
        public string TrainingEventBusinessUnitAcronym { get; set; }
        public Boolean HasHits { get; set; }
        public int TotalHits { get; set; }
        public string SubmittedAppUserFirstName { get; set; }
        public string SubmittedAppUserLastName { get; set; }
        public string AcceptedAppUserFirstName { get; set; }
        public string AcceptedAppUserLastName { get; set; }
        public string CourtesyCompleteAppUserFirstName { get; set; }
        public string CourtesyCompleteAppUserLastName { get; set; }
        public string SentToCourtesyAppUserFirstName { get; set; }
        public string SentToCourtesyAppUserLastName { get; set; }
        public string SentToLeahyAppUserFirstName { get; set; }
        public string SentToLeahyAppUserLastName { get; set; }
        public long FileID { get; set; }
        public int NumOfRemovedParticipants { get; set; }
        public int NumOfCanceledParticipants { get; set; }

        public ICourtesyBatch_Item CourtesyBatch { get; set; }
        public List<IPersonVetting_Item> PersonVettings { get; set; }
        public List<IPersonVettingType_Item> PersonVettingTypes { get; set; }
        public List<IPersonVettingHit_Item> PersonVettingHits { get; set; }
        public List<IPersonVettingVettingType_Item> PersonVettingVettingTypes { get; set; }
    }
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.VettingService.Models/vetting-batch_item.ts


import { ICourtesyBatch_Item } from './icourtesy-batch_item';
import { IPersonVetting_Item } from './iperson-vetting_item';
import { IPersonVettingType_Item } from './iperson-vetting-type_item';
import { IPersonVettingHit_Item } from './iperson-vetting-hit_item';
import { IPersonVettingVettingType_Item } from './iperson-vetting-vetting-type_item';
import { IVettingBatch_Item } from './ivetting-batch_item';

export class VettingBatch_Item implements IVettingBatch_Item {
  
	public Ordinal: number = 0;
	public VettingBatchID: number = 0;
	public VettingBatchName: string = "";
	public VettingBatchNumber: number = 0;
	public VettingBatchOrdinal: number = 0;
	public TrainingEventID?: number;
	public CountryID?: number;
	public CountryName: string = "";
	public VettingBatchTypeID: number = 0;
	public VettingBatchType: string = "";
	public AssignedToAppUserID?: number;
	public AssignedToAppUserFirstName: string = "";
	public AssignedToAppUserLastName: string = "";
	public VettingBatchStatusID: number = 0;
	public VettingBatchStatus: string = "";
	public IsCorrectionRequired: boolean = false;
	public CourtesyVettingOverrideFlag?: boolean;
	public CourtesyVettingOverrideReason: string = "";
	public GTTSTrackingNumber: string = "";
	public LeahyTrackingNumber: string = "";
	public INKTrackingNumber: string = "";
	public DateVettingResultsNeededBy?: Date;
	public DateSubmitted?: Date;
	public DateAccepted?: Date;
	public DateSentToCourtesy?: Date;
	public DateCourtesyCompleted?: Date;
	public DateSentToLeahy?: Date;
	public DateLeahyResultsReceived?: Date;
	public DateVettingResultsNotified?: Date;
	public EventStartDate?: Date;
	public DateLeahyFileGenerated?: Date;
	public VettingFundingSourceID: number = 0;
	public VettingFundingSource: string = "";
	public AuthorizingLawID: number = 0;
	public AuthorizingLaw: string = "";
	public VettingBatchNotes: string = "";
	public ModifiedByAppUserID: number = 0;
	public ModifiedDate: Date = new Date(0);
	public TrainingEventName: string = "";
	public TrainingEventBusinessUnitAcronym: string = "";
	public HasHits: boolean = false;
	public TotalHits: number = 0;
	public SubmittedAppUserFirstName: string = "";
	public SubmittedAppUserLastName: string = "";
	public AcceptedAppUserFirstName: string = "";
	public AcceptedAppUserLastName: string = "";
	public CourtesyCompleteAppUserFirstName: string = "";
	public CourtesyCompleteAppUserLastName: string = "";
	public SentToCourtesyAppUserFirstName: string = "";
	public SentToCourtesyAppUserLastName: string = "";
	public SentToLeahyAppUserFirstName: string = "";
	public SentToLeahyAppUserLastName: string = "";
	public FileID: number = 0;
	public NumOfRemovedParticipants: number = 0;
	public NumOfCanceledParticipants: number = 0;
	public CourtesyBatch?: ICourtesyBatch_Item;
	public PersonVettings?: IPersonVetting_Item[];
	public PersonVettingTypes?: IPersonVettingType_Item[];
	public PersonVettingHits?: IPersonVettingHit_Item[];
	public PersonVettingVettingTypes?: IPersonVettingVettingType_Item[];
  
}


